core/domain/story_services.py

```diff
@@ -201,30 +201,56 @@ def _save_story(committer_id, story, commit_message, change_list):
     story.validate()
     # Validate that all explorations referenced by the story exist.
     exp_ids = []
+
+    # Any exp ID in the story to compare categories later on.
+    exp_id = None
     for node in story.story_contents.nodes:
         if node.exploration_id is not None:
             exp_ids.append(node.exploration_id)
-    exp_summaries = (
-        exp_fetchers.get_exploration_summaries_matching_ids(exp_ids))
+            if exp_id is None:
+                exp_id = node.exploration_id
+
+    # Strict = False, since the existence of explorations is checked below.
+    exps_dict = (
+        exp_fetchers.get_multiple_explorations_by_id(exp_ids, strict=False))
 
-    exp_summaries_dict = {
-        exp_id: exp_summaries[ind] for (ind, exp_id) in enumerate(exp_ids)
-    }
     for node in story.story_contents.nodes:
         if (node.exploration_id is not None) and (
-                not exp_summaries_dict[node.exploration_id]):
+                node.exploration_id not in exps_dict):
             raise utils.ValidationError(
                 'Expected story to only reference valid explorations, '
                 'but found an exploration with ID: %s (was it deleted?)' %
                 node.exploration_id)
 
-    if exp_summaries:
-        common_exp_category = exp_summaries[0].category
-        for summary in exp_summaries:
-            if summary.category != common_exp_category:
+    valid_interaction_ids = [
```

Comment:
Add a detailed comment here that explains the criterion for "valid", so that if it changes in future we can understand what the constraints on such a change are.

Reply:
Done